harpy/_validations.py

```
@@ -11,14 +11,34 @@
 import rich_click as click
 from ._printing import print_error, print_notice, print_solution, print_solution_with_culprits
+from ._misc import harpy_progressbar
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+def is_gzip(file_path):
+    """helper function to determine if a file is gzipped"""
+    try:
+        with gzip.open(file_path, 'rt') as f:
+            f.read(10)
+        return True
+    except:
```

Ruff

Do not use bare `except`

(E722)